### PR TITLE
Add support to read registry artifacts in MI

### PR DESCRIPTION
**Pull request by chanikag**
## Purpose
This is to resolve registry path in dataservices in sql driver

### CODE REVIEW

File: components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/TDriverUtil.java [hunk 1]
```diff
@@ -318,37 +328,18 @@ public static InputStream getInputStreamFromPath(String path) throws IOException
...
-//                } else {
-//                    throw new SQLException(
-//                            "The given XSLT resource path at '" + path + "' does not exist");
-//                }
-//            } catch (RegistryException e) {
-//                throw new SQLException(e);
-//            }
+            ConfigurationContext configCtx = SQLDriverDSComponent.getConfigurationContextService().
+                    getServerConfigContext();
+            if (configCtx != null) {
```
Comment (sdkottegoda) [inline components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/TDriverUtil.java]: Can this be null?

File: components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/internal/SQLDriverDSComponent.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+* @scr.reference name="configContext.service" interface="org.wso2.micro.integrator.core.services.Axis2ConfigurationContextService"
+* cardinality="1..1" policy="dynamic"  bind="setAxis2ConfigurationContextService" unbind="unsetAxis2ConfigurationContextService"
+*/
+public class SQLDriverDSComponent {
+
+    private static Log log = LogFactory.getLog(SQLDriverDSComponent.class);
+
+    private static Axis2ConfigurationContextService configurationContextService = null;
+
+    public SQLDriverDSComponent() {
```
Comment (sdkottegoda) [inline components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/internal/SQLDriverDSComponent.java]: We don't need to add the implicit public constructor

File: components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/internal/SQLDriverDSComponent.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.micro.integrator.core.services.Axis2ConfigurationContextService;
+
+/**
+* @scr.component name="org.wso2.carbon.dataservices.sql.driver" immediate="true"
+* @scr.reference name="configContext.service" interface="org.wso2.micro.integrator.core.services.Axis2ConfigurationContextService"
+* cardinality="1..1" policy="dynamic"  bind="setAxis2ConfigurationContextService" unbind="unsetAxis2ConfigurationContextService"
```
Comment (sdkottegoda) [inline components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/internal/SQLDriverDSComponent.java]: Older annotation is not compatible with the newer version OSGI version

File: components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/TDriverUtil.java (hunk 1 above)
Comment (chanikag) [reply components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/TDriverUtil.java]: Removed null check

File: components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/internal/SQLDriverDSComponent.java (hunk 2 above)
Comment (chanikag) [reply components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/internal/SQLDriverDSComponent.java]: Removed

File: components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/internal/SQLDriverDSComponent.java (hunk 3 above)
Comment (chanikag) [reply components/data/data-services/org.wso2.micro.integrator.dataservices.sql.driver/src/main/java/org/wso2/micro/integrator/dataservices/sql/driver/internal/SQLDriverDSComponent.java]: Moved to new annotation. But this does not give any compilation errors/ runtime errors

**Pull request merged**